var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -283,8 +284,20 @@ def filter_detected_exes(cls, prefix, exes_in_prefix):
             # clang++ matches g++ -> clan[g++]
             if any(x in exe for x in ('clang', 'ranlib')):
                 continue
-            # Filter out links in favor of real executables
-            if os.path.islink(exe):
+            # Filter out links in favor of real executables on
+            # all systems but Cray
+            host_platform = str(spack.architecture.platform())
+            if os.path.islink(exe) and host_platform != 'cray':
+                continue
+            # Apple's gcc is actually apple clang, so skip it.
```

Comment:
I think this comment needs to move a few lines down to the `if` statement which checks for `"Apple"`